### PR TITLE
Test restart

### PR DESCRIPTION
**Pull request by badurinantun**
Tests for restart method

### CODE REVIEW

File: test/devnet-restart.test.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+describe.only("Devnet restart", function() {
+    this.timeout(TIMEOUT);
+
+    const salt = "0x99";
+
+    it("should pass", async () => {
+        try {
+            const response = await starknet.devnet.restart();
+            expect(response).to.be.undefined;
+        } catch (error) {
```
Comment (FabijanC) [inline test/devnet-restart.test.ts:15]: Why catch this? If it throws, it already makes the test fail. By failing like this, the real failure cause is hidden.

File: test/devnet-restart.test.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        let contract = await contractFactory.deploy({ initial_balance: 0 }, { salt });
+        const initialAddress = contract.address;
+
+        starknet.devnet.restart();
+
+        try {
+            contract = await contractFactory.deploy({ initial_balance: 0 }, { salt });
+            expect(contract.address).to.equal(initialAddress);
+        } catch {
+            expect.fail("Should not throw");
```
Comment (FabijanC) [inline test/devnet-restart.test.ts:49]: As noted above.

File: test/devnet-restart.test.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        // redeploy
+        await contractFactory.deploy({ initial_balance: 0 }, { salt });
+        const { res: balanceAfterRestart } = await contract.call("get_balance");
+        expect(balanceAfterRestart).to.deep.equal(0n);
+
+        try {
+            await contract.invoke("increase_balance", { amount1: 10, amount2: 20 });
+            expect(contract.address).to.equal(initialAddress);
+        } catch {
+            expect.fail("Should not throw");
```
Comment (FabijanC) [inline test/devnet-restart.test.ts:73]: As noted above.

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: test/devnet-restart.test.ts (hunk 3 above)
Comment (badurinantun) [reply test/devnet-restart.test.ts]: [d9788ec](https://github.com/Shard-Labs/starknet-hardhat-example/pull/49/commits/d9788ec7f843ac9bb798c14199a7501c5516ea95)

File: test/devnet-restart.test.ts (hunk 2 above)
Comment (badurinantun) [reply test/devnet-restart.test.ts]: [d9788ec](https://github.com/Shard-Labs/starknet-hardhat-example/pull/49/commits/d9788ec7f843ac9bb798c14199a7501c5516ea95)

File: test/devnet-restart.test.ts (hunk 1 above)
Comment (badurinantun) [reply test/devnet-restart.test.ts]: [d9788ec](https://github.com/Shard-Labs/starknet-hardhat-example/pull/49/commits/d9788ec7f843ac9bb798c14199a7501c5516ea95)

**Review by FabijanC — APPROVED** [review]

**Pull request merged**